src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java

```diff
@@ -148,14 +148,12 @@ private synchronized static CompletableFuture<Void> internalUpdateClasspathWithQ
         }
         var project = module.getProject();
         final CompletableFuture<Void> future = new CompletableFuture<>();
-        CompletableFuture.runAsync(() -> {
-            Runnable task = () -> ProgressManager.getInstance().run(new Task.Backgroundable(project, "Adding Quarkus deployment dependencies to classpath...") {
+        MavenUtil.invokeAndWait(project, () -> ProgressManager.getInstance().run(new Task.Backgroundable(project, "Adding Quarkus deployment dependencies to classpath...") {

```

Comment:
Your code seems blocking the Thread 'since you are using invokeAndWait. The current code uses runAsynch.

Reply:
reverted this change.